frontend/pages/home/components/allowance/EditForm.tsx

```diff
@@ -0,0 +1,205 @@
+import { middleTruncation } from "@/utils/strings";
+import Button from "@components/buttons/Button";
+import { Allowance, ErrorFields } from "@/@types/allowance";
+import { Asset } from "@redux/models/AccountModels";
+import { CurrencyInput } from "@components/input";
+import { getAssetIcon } from "@/utils/icons";
+import { IconTypeEnum } from "@/const";
+import { useMemo } from "react";
+import dayjs from "dayjs";
+import { useUpdateAllowance } from "@pages/home/hooks/useUpdateAllowance";
+import { ValidationErrors } from "@/@types/common";
+import { Chip } from "@components/chip";
+import { CheckBox } from "@components/checkbox";
+import { CalendarPicker } from "@components/CalendarPicker";
+import clsx from "clsx";
+import { validationMessage } from "@/helpers/schemas/allowance";
+
+export default function UpdateForm() {
```

Comment:
This file is too big, please lets split it in different files.

Reply:
Done